src/Main.cc

```diff
@@ -48,7 +48,7 @@ int main(int argc, const char **argv)
     spdlog::info("{}: Successfully added.", exName);
 
     const auto exAttempt = parser().get("sources");
-    if (exercises.testExercise(exName, exAttempt) != 0)
+    if (!exercises.testExercise(exName, exAttempt).has_value())
```

Comment:
Optionals have an implicit conversion to booleans.

```suggestion
    if (!exercises.testExercise(exName, exAttempt))
```